src/app/core/config-ui/config-entity/config-entity.component.ts

```diff
@@ -0,0 +1,135 @@
+    private location: Location,
+    private entityMapper: EntityMapperService,
+    private entityActionsService: EntityActionsService,
+  ) {}
+
+  ngOnInit(): void {
+    this.init();
+  }
+
+  private init() {
+    this.entityConstructor = this.entities.get(this.entityType);
+    this.originalEntitySchemaFields = JSON.parse(
+      JSON.stringify(Array.from(this.entityConstructor.schema.entries())),
+    );
+
+    const detailsView: ViewConfig<EntityDetailsConfig> =
+      this.configService.getConfig(
+        EntityConfigService.getDetailsViewId(this.entityConstructor),
+      );
+    if (detailsView.component !== "EntityDetails") {
+      // not supported currently
+      return;
+    }
+
+    // work on a deep copy as we are editing in place (for titles, sections, etc.)
+    this.configDetailsView = JSON.parse(JSON.stringify(detailsView.config));
+  }
+
+  cancel() {
+    this.entityConstructor.schema = new Map(this.originalEntitySchemaFields);
+    this.location.back();
+  }
+
+  async save() {
+    const originalConfig = await this.entityMapper.load(
+      Config,
+      Config.CONFIG_KEY,
+    );
+    const newConfig = originalConfig.copy();
+
+    this.setViewConfig(newConfig);
+    this.setEntityConfig(newConfig);
+
+    await this.entityMapper.save(newConfig);
+    this.entityActionsService.showSnackbarConfirmationWithUndo(
+      newConfig,
+      $localize`:Save config confirmation message:updated`,
+      [originalConfig],
+    );
+
+    this.location.back();
+  }
+
+  private setViewConfig(newConfig: Config) {
+    const viewId = EntityConfigService.getDetailsViewId(this.entityConstructor);
+    newConfig.data[viewId].config = this.configDetailsView;
+  }
+
+  private setEntityConfig(newConfig: Config) {
+    const entityConfigKey =
+      EntityConfigService.PREFIX_ENTITY_CONFIG + this.entityType;
+
+    // init config if not present
+    newConfig.data[entityConfigKey] =
+      newConfig.data[entityConfigKey] ?? ({ attributes: {} } as EntityConfig);
+    newConfig.data[entityConfigKey].attributes =
+      newConfig.data[entityConfigKey].attributes ?? {};
+
+    const entitySchemaConfig: EntityConfig = newConfig.data[entityConfigKey];
+
+    for (const [fieldId, field] of this.entityConstructor.schema.entries()) {
+      if (!field._isCustomizedField) {
+        // do not write unchanged default fields from the classes into config
+        continue;
+      }
+      entitySchemaConfig.attributes[fieldId] = field;
+    }
+  }
+
+  createPanel() {
+    const newPanel: Panel = { title: "New Tab", components: [] };
+    this.configDetailsView.panels.push(newPanel);
+
+    // wait until view has actually added the new tab before we can auto-select it
+    setTimeout(() => {
+      const newTabIndex = this.configDetailsView.panels.length - 1;
+      this.tabGroup.selectedIndex = newTabIndex;
+      this.tabGroup.focusTab(newTabIndex);
+    });
+  }
+
+  addSection(panel: Panel) {
+    panel.components.push({
```

Comment:
This does not seem to work yet. The UI is broken and the form is not editable. I guess its missing a expected initial config.